craft_application/services/lifecycle.py

```diff
@@ -425,7 +425,7 @@ def _validate_build_plan(build_plan: list[models.BuildInfo]) -> None:
         raise errors.EmptyBuildPlanError
 
     if len(build_plan) > 1:
-        raise errors.MultipleBuildsError
+        raise errors.MultipleBuildsError(matching_builds=build_plan)
```

Comment:
The code looks good, but please create a unit test that fires this off and checks the error message.

Reply:
Of course, good call!